model/user_task_annotations/user_task_annotations.go

```diff
@@ -0,0 +1,18 @@
+package user_task_annotations
+
+import "github.com/evergreen-ci/evergreen/model/task_annotations"
+
+// UserTaskAnnotation will store any edits that a user makes to a task annotation
+type UserTaskAnnotation struct {
```

Comment:
Will this always exactly match TaskAnnotation? You could just embed TaskAnnotation rather than all this copy/paste. (example:) https://github.com/evergreen-ci/evergreen/blob/28d206c77af2fad21ba09bf667dc7eaf29e37e4c/model/project.go#L251

Reply:
it seems more likely this will be a subset of TaskAnnotation (it's already doing without the metadata field).

Reply:
Ah then vise versa :) Although I do wonder if we need separate document types at all--can this just use the same struct but store it in two different collections? Or should there be a field in the document that denotes where it's from and that can be an indexed field for just the one collection? Or do we expect these structs to diverge even more in the future and thus we want to allow for that?

Reply:
The information that these two structs hold will be completely different. However, the only difference in the structure right now is that one holds metadata and one doesn't. If we use the same struct, what would be the benefit of storing it in two different collections?

Reply:
"_should there be a field in the document that denotes where it's from_"
We can technically do that, but then the user sourced annotation will always have an empty metadata field. If we want to take that approach, something like this might make more sense  https://github.com/malikchaya2/evergreen/pull/new/EVG-13333-A.

Reply:
I don't think there's anything wrong with always having an empty field for one kind of annotation--this is true for different kinds of patches, different kinds of commit queues, etc.

Reply:
The only other concern I'd have with the one struct two collections approach is that it would need to allow for more than one task annotation per taskID, but we only want one task_annotation and one user_task_annotation per task (with one being overwritten whenever there's an update).

Reply:
> use the same struct but store it in two different collections

I would think this would make for awkward CRUD operations: which collection would `func (a *TaskAnnotation) Insert()`  use?